### PR TITLE
Pin python>=3.6.1

### PR DESCRIPTION
**Pull request by davidbrochart**
See #634

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -63,10 +63,10 @@ def run(self):
         'Operating System :: OS Independent',
         'Programming Language :: Python',
         'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
         'Programming Language :: Python :: 3.8',
+        'Programming Language :: Python :: 3.9',
```
Comment (mriedem) [inline setup.py:69]: Unrelated to this py3.5 change but just confirming that py3.9 is being tested:

https://github.com/jupyter/jupyter_client/blob/018d8da0b947865d8217b459197bcc4f13e04666/.github/workflows/main.yml#L41

👍

File: setup.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def run(self):
         'tornado>=4.1',
         'nest-asyncio>=1.5',
     ],
-    python_requires='>=3.5',
+    python_requires='>=3.6.1',
```
Comment (mriedem) [inline setup.py:79]: So anyone using 3.5 with 6.1.13 will still have issues. What's the plan there? Does 6.1.13 get revoked/blacklisted somehow and then _this_ change goes into a new 6.2 release?

**Review by mriedem — COMMENTED** [review]
Looks OK but a question inline.

File: setup.py (hunk 2 above)
Comment (davidbrochart) [reply setup.py:79]: Yes, v6.1.13 has been marked as broken on conda-forge, and I'm waiting for someone with management rights to yank it on PyPI.
When we merge this PR, I will release v6.2.

**Review by mriedem — APPROVED** [review]
LGTM

File: setup.py (hunk 2 above)
Comment (MSeal) [reply setup.py:79]: @davidbrochart I can yank 6.1.13. Do you want me to wait for the 6.2 release before yanking? Usually you want a roll-forward option before removing a dep, especially when it's not a security issue.

**Review by MSeal — APPROVED** [review]

File: setup.py (hunk 2 above)
Comment (MSeal) [reply setup.py:79]: @davidbrochart 6.1.13 is yanked. You should have permission on PyPI for this as well (let me know if you don't see the option)

Comment (davidbrochart) [reply setup.py:79]: Thanks Matthew. Unfortunately I still don't see admin permission on PyPI.

Comment (MSeal) [reply setup.py:79]: Aww I think I see the issue. Should be fixed now.

Comment (davidbrochart) [reply setup.py:79]: Yes it is, thanks!

**Pull request merged**